simple-schema.js

```diff
@@ -835,11 +835,14 @@ SimpleSchema.prototype.getDefinition = function(key, propList, functionContext)
 
   // For any options that support specifying a function,
   // evaluate the functions.
-  _.each(['min', 'max', 'minCount', 'maxCount', 'allowedValues', 'optional', 'label'], function (prop) {
+  _.each(['min', 'max', 'minCount', 'maxCount', 'allowedValues', 'label'], function (prop) {
     if (_.isFunction(defs[prop])) {
       defs[prop] = defs[prop].call(functionContext || {});
     }
   });
+  if (functionContext && _.isFunction(defs['optional'])) {
+    defs['optional'] = defs['optional'].call(functionContext || {});
+  }
```

Comment:
@aldeed 

I will make a breaking change if I do what you suggested. `getDefinition` is called in `messageForError` function for other props. I can't create a context there. Again I may be missing something?

```js
// Returns a string message for the given error type and key. Uses the
// def and value arguments to fill in placeholders in the error messages.
SimpleSchema.prototype.messageForError = function(type, key, def, value) {
  var self = this;
  
  // We proceed even if we can't get a definition because it might be a keyNotInSchema error
  def = def || self.getDefinition(key, ['regEx', 'label', 'minCount', 'maxCount', 'min', 'max', 'type']) || {};
```

Regarding context. Yes I can and will create it earlier in the function and then pass it to both calls.